jaxrs/src/main/java/org/jboss/as/jaxrs/JaxrsSubsystemParser_2_0.java

```diff
@@ -0,0 +1,132 @@
+
+import javax.xml.stream.XMLStreamConstants;
+import javax.xml.stream.XMLStreamException;
+
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.controller.parsing.ParseUtils;
+import org.jboss.as.controller.persistence.SubsystemMarshallingContext;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.jboss.staxmapper.XMLElementReader;
+import org.jboss.staxmapper.XMLElementWriter;
+import org.jboss.staxmapper.XMLExtendedStreamReader;
+import org.jboss.staxmapper.XMLExtendedStreamWriter;
+
+public class JaxrsSubsystemParser_2_0 implements XMLStreamConstants, XMLElementReader<List<ModelNode>>, XMLElementWriter<SubsystemMarshallingContext> {
+
+   private static final Pattern PATTERN_LIST = Pattern.compile(".+(,.+)*");
+   private static final Pattern PATTERN_PROPERTY_LIST = Pattern.compile(".+:.+(,.+:.+)*");
+
+   /**
+    * {@inheritDoc}
+    */
+   @Override
+   public void readElement(final XMLExtendedStreamReader reader, final List<ModelNode> list) throws XMLStreamException {
+       // Require no attributes or content
+       requireNoAttributes(reader);
+       list.add(Util.createAddOperation(PathAddress.pathAddress(JaxrsExtension.SUBSYSTEM_PATH)));
+
+       //Read the children
+       while (reader.hasNext() && reader.nextTag() != END_ELEMENT) {
+           if (!reader.getLocalName().equals("context-parameters")) {
+               throw ParseUtils.unexpectedElement(reader);
+           }
+           while (reader.hasNext() && reader.nextTag() != END_ELEMENT) {
+               if (reader.isStartElement()) {
+                   readContextParameter(reader, list);
+               }
+           }
+       }
+   }
+
+   private void readContextParameter(XMLExtendedStreamReader reader, List<ModelNode> list) throws XMLStreamException {
+      if (!reader.getLocalName().equals("context-parameter")) {
+          throw ParseUtils.unexpectedElement(reader);
+      }
+
+      String name = reader.getAttributeValue(0);
+      String value = reader.getAttributeValue(1);
+      if (PATTERN_PROPERTY_LIST.matcher(value).matches() || PATTERN_LIST.matcher(value).matches()) {
+         String[] ss = value.split(",");
+         ModelNode modelNode = new ModelNode();
+         for (String s : ss) {
+            modelNode.add(s.trim());
+         }
+         JaxrsContextParamHandler.getContextParameters().put(name, modelNode);
+      } else {
+         JaxrsContextParamHandler.getContextParameters().put(name, new ModelNode(value));
+      }
+      ParseUtils.requireNoContent(reader);
+  }
+
+   /**
+    * {@inheritDoc}
+    */
+   @Override
+   public void writeContent(final XMLExtendedStreamWriter streamWriter, final SubsystemMarshallingContext context) throws XMLStreamException {
+       context.startSubsystemElement(JaxrsExtension.NAMESPACE_2_0, false);
+       Map<String, ModelNode> contextParameters = JaxrsContextParamHandler.getContextParameters();
```

Comment:
I think we should just use what is in the model when writing this out. I mentioned using a service to store this state rather than a static field, and that service can only be populated in a normal running server.

If someone starts the service with --admin-only, there is no runtime stage for this service to get installed. And we need to be able to do the marshalling when running in admin-only